python/doc/examples/meta_modeling/kriging_metamodel/plot_kriging_categorical.py

```diff
@@ -0,0 +1,375 @@
+# ProductCovarianceModel class. We start here by defining the product kernel,
+# which combines SquaredExponential kernels for the continuous variables, and
+# LatentVariableModel for the discrete ones.
+
+# %%
+latdim = 2  # Dimension of the latent space
+activecoord = (
+    2 + latdim * (lvls1 - 2) + latdim * (lvls2 - 2)
+)  # Nb ative coordinates in the latent space
+kx1 = ot.SquaredExponential(1)
+kx2 = ot.SquaredExponential(1)
+kz1 = otexp.LatentVariableModel(lvls1, latdim)
+kz2 = otexp.LatentVariableModel(lvls2, latdim)
+kLV = ot.ProductCovarianceModel([kx1, kx2, kz1, kz2])
+kLV.setNuggetFactor(1e-6)
+# Bounds for the hyperparameter optimization
+lb_LV = [1e-4] * dim + [-10] * activecoord
+ub_LV = [3.] * dim + [10.] * activecoord
+bounds_LV = ot.Interval(lb_LV, ub_LV)
+# Distribution for the hyperparameters initialization
+initdist_LV = ot.DistributionCollection()
+for i in range(len(lb_LV)):
+    initdist_LV.add(ot.Uniform(lb_LV[i], ub_LV[i]))
+initdist_LV = ot.ComposedDistribution(initdist_LV)
+
+# %%
+# Alternatively, we consider a purely continuous kernel for independent Gaussian processes.
+# One for each combination of categorical variables levels.
+
+# %%
+k_ind = ot.SquaredExponential(2)
+lb_ind = [1e-4, 1e-4]
+ub_ind = [3., 3.]
+bounds_ind = ot.Interval(lb_ind, ub_ind)
+initdist_ind = ot.DistributionCollection()
+for i in range(len(lb_ind)):
+    initdist_ind.add(ot.Uniform(lb_ind[i], ub_ind[i]))
+initdist_ind = ot.ComposedDistribution(initdist_ind)
+initSample_ind = initdist_ind.getSample(10)
+optalg_ind = ot.MultiStart(ot.NLopt("LN_COBYLA"), initSample_ind)
+
+# %%
+# In order to assess their respective robustness with regards to the training data set,
+# we repeat the experiments 10 times with different training of size 72,
+# and compute each time the normalized prediction Root Mean Squared Error (RMSE) on a
+# test data set # of size 1000.
+
+# %%
+RMSE_LV = []
+RMSE_ind = []
+for rep in range(5):
+    # We generate the normalized training data set
+    X = dist.getSample(72)
+    Y = fun(X)
+    Ymax = Y.getMax()
+    Ymin = Y.getMin()
+    Y = (Y - Ymin) / (Ymin - Ymax)
+
+    # Initialize  and parameterize the optimization algorithm
+    initSample_LV = initdist_LV.getSample(10)
+    optalg_LV = ot.MultiStart(ot.NLopt("LN_COBYLA"), initSample_LV)
+
+    # Create and train the Gaussian process models
+    basis = ot.ConstantBasisFactory(dim).build()
+    algo_LV = ot.KrigingAlgorithm(X, Y, kLV, basis)
+    algo_LV.setOptimizationAlgorithm(optalg_LV)
+    algo_LV.setOptimizationBounds(bounds_LV)
+    algo_LV.run()
+    res_LV = algo_LV.getResult()
+
+    # Compute the models predictive performances on a validation data set
+    Xval = dist.getSample(1000)
+    Yval = fun(Xval)
+    Yval = (Yval - Ymin) / (Ymin - Ymax)
+
+    Val_LV = ot.MetaModelValidation(Xval, Yval, res_LV.getMetaModel())
+    rmse_LV = np.sqrt(np.mean(np.array(Val_LV.getResidualSample()) ** 2))
```

Comment:
Up to the change from $n$ to $n - 1$, this looks like [Sample.computeStandardDeviation()](http://openturns.github.io/openturns/latest/user_manual/_generated/openturns.Sample.html#openturns.Sample.computeStandardDeviation), isn'it?

Reply:
It is similar, but on a finite sample the average residual will not be null.

Reply:
I do not understand your answer. Moreover, I guess that this reveals that a new MetaModelValidation.computeRootMeanSquaredError() method should be created. This is out of the scope of this PR though.

Reply:
I created MetaModelValidation.computeMeanSquaredError() at https://github.com/openturns/openturns/pull/2330/files#diff-8eb6060d1ee247fcf98f5f7fcb0a59655266d95063bb3f27f0d58f1d4d30531d. Should I create MetaModelValidation.computeRootMeanSquaredError() so that this particular example can be simplified?